src/Confluent.Kafka/Serialization/NullDeserializer.cs

```diff
@@ -16,8 +16,20 @@
 
 namespace Confluent.Kafka.Serialization
 {
+    /// <summary>
+    ///     A 'dummy' deserializer for use with values that must be null.
+    /// </summary>
     public class NullDeserializer : IDeserializer<Null>
     {
+        /// <summary>
+        ///     'deserializes' any byte array to a null value.
+        /// </summary>
+        /// <param name="data">
+        ///     The data to deserialize (should be null)
```

Comment:
should was more correct because there was previously no check that the input data was `null`. However I just added this check - I think it's a valuable check that will catch real world errors. This is technically an API change (change in semantics), but I propose just making it.